docs/advanced/signal_generators.md

```diff
@@ -0,0 +1,638 @@
+# Signal Generation
+
+An overview of the signal generation functionality available in this package, along with information about Tektronix signal generators.
+
+---
+
+## Function Generation
+
+A function is a limited set of common waveforms that are provided by default through the instrument.
+The simplicity of these waveforms allows for output parameters like waveform length and sample rate to be
+abstracted away. Frequency replaces these in order to provide signals which are easy to quantify and manipulate.
+
+Arbitrary Function Generators (\[AFGs\](default: AFG)) utilize a phase increment process and a data lookup table to provide variable frequencies.
```

Comment:
```suggestion
Arbitrary Function Generators ([AFGs](default: AFG)) utilize a phase increment process and a data lookup table to provide variable frequencies.
```

Reply:
All the square brackets seem to have been escaped. That will render them all useless.